custom/icds_reports/management/commands/national_user_data_pull.py

```diff
@@ -0,0 +1,65 @@
+import csv
+
+from django.core.management.base import BaseCommand
+from corehq.apps.users.models import CommCareUser, WebUser
+from custom.icds_reports.const import INDIA_TIMEZONE
+from custom.icds_reports.models import ICDSAuditEntryRecord
+from django.db.models import Max
+from dimagi.utils.chunked import chunked
+
+
+class Command(BaseCommand):
+    help = "List of users with national level access on Dashboard along with usage"
+
+    def convert_to_ist(self, date):
+        if date is None:
+            return 'N/A'
+        date = date.astimezone(INDIA_TIMEZONE)
+        date_formatted = date.strftime("%d-%m-%Y")
+        return date_formatted
+
+    def _get_details(self, users):
+        """
+        :param users: user objects of all commcare and web users
+        :return: user_details dict
+        """
+        user_details = {}
+        for user in users:
+            role = user.get_role('icds-cas')
+            if (role in ('CPMU', 'Dashboard Only Access', 'TRP') or len(user.assigned_location_ids) == 0) and user.has_permission('icds-cas', 'access_all_locations'):
+                user_details.update({user.username: [user.created_on, role.name]})
+        return user_details
+
+    def handle(self, *args, **options):
+        users = CommCareUser.by_domain('icds-cas')
+        web_users = WebUser.by_domain('icds-cas')
+        users = users + web_users
+        user_details = self._get_details(users)
+        usernames_list = [*user_details]
```

Comment:
`usernames_list = list(user_details.keys())` would be easier to understand. At first glance, I was not able to parse whats happening here.

Reply:
updated